### ISSUE TITLE
MySqlConnector.MySqlException (0x80004005): This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'

### ISSUE THREAD

**Issue by lujie19930803** [elsa-workflows/elsa-core#6219]
## Description
when i use mysql to perisist data. i meet the following error.  and my elsa version is 3.2.2. mysql version is 8.0.29.
![image](https://github.com/user-attachments/assets/21cb0e79-4c23-47ed-b668-2a6c12c73f1a)

## Steps to Reproduce
i configure elsa like that.
```
services.AddElsa(elsa =>
        {
            elsa.UseWorkflowRuntime(runtime => runtime.UseEntityFrameworkCore(ef => ef.UseMySql(connectionStr)));
            elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore(ef => ef.UseMySql(connectionStr)));
            elsa.UseIdentity(identity =>
            {
                identity.TokenOptions = options => options.SigningKey = "sufficiently-large-secret-signing-key"; // This key needs to be at least 256 bits long.
                identity.UseAdminUserProvider();
            });
            elsa.UseDefaultAuthentication(auth => auth.UseAdminApiKey());
            elsa.UseWorkflowsApi();
            elsa.UseRealTimeWorkflows();
            elsa.UseCSharp();
            elsa.UseJavaScript();
            elsa.UseFileStorage();
            elsa.UseHttp();
            elsa.UseScheduling();
        });
```

i found when i comment UseWorkflowRuntime.  this error disappeared.
![image](https://github.com/user-attachments/assets/5f465f74-77ea-442e-ad90-5f4ed85b48e6)
@sfmskywalker please help me.

**Comment by r-kernberger** [comment]
@lujie19930803 I had the same error with v3.2. Give v3.3.0-rc3 a try. It was fixed for me then

**Comment by sfmskywalker** [comment]
It sounds familiar to what is happening in #6126

If the issue persists despite @r-kernberger 's suggestion, here are two other options to consider:

1. Upgrade to the latest MySql (>= v9) version.
2. Generate EF Core migrations manually and use them instead.

The MySql EF Core migrations shipped with Elsa target the latest MySql version and are included for convenience, however these are not required. Some users even prefer to maintain their own migrations.

**Comment by sfmskywalker** [comment]
I performed some tests with MySQL 8.0.0 and Elsa running on .NET 8, and I ran into some issues.
I was able to work around those issues by enabling primitive support and automatically translating parameterized collections to constants, as [explained here](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1960#issue-2738728356). 

After that, the MySql provider seems to work well, so far. I'll conduct more tests.
